hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/ProbeBuilderImpl.java

```diff
@@ -48,16 +48,28 @@ private ProbeBuilderImpl(MetricsRegistryImpl metricsRegistry, String keyPrefix,
     @Override
     @CheckReturnValue
     public ProbeBuilderImpl withTag(String tag, String value) {
+        return withTag(tag, value, false);
+    }
+
+    private ProbeBuilderImpl withMetricTag(String metricName) {
+        return withTag("metric", prefixedMetricName(metricName), true);
```

Comment:
I think we should keep the `name` as the first metric. The metric compressor skips the first N chars that are identical to the first N chars of the last recorded metric. So by thinking more about this, we have two sorting goals here:
1) One that supports metric compressor. I think the order supports it the most is `name,unit,metric` (what was initially the order).
2) One that improves the readability of the diagnostics logs with `name,metric,unit`.

It feels now that the sorting should be the concern of the `MetricsPlugin` after tokenizing the tags.

Reply:
> One that improves the readability of the diagnostics logs with `name,metric,unit`.

Shouldn't it be something like `metric,name,unit` instead of `name,metric,unit`?

Reply:
If we sort for readability, I think we should group by the metric's source first and then sort by the metrics, hence I think `name` should be the first. But I think this should be addressed outside of this PR, I'd revert the last commit.

Reply:
Decided to postpone sort related changes until the code base gets more stable. Thus, I've reverted the corresponding commit.